kubectl-openebs/cli/command/describe/pvc_info.go

```diff
@@ -0,0 +1,268 @@
+			// Get the casType from the storage class and branch on basic of CSTOR and NON-CSTOR PVCs.
+			casType := client.GetCasTypeFromSC(sc)
+			if casType == util.CstorCasType {
+
+				// Create Empty template objects and fill gradually when underlying sub CRs are identified.
+				pvcInfo := util.CstorPVCInfo{}
+				cvcInfo := util.CVCInfo{}
+
+				pvcInfo.Name = item.Name
+				pvcInfo.Namespace = item.Namespace
+				pvcInfo.BoundVolume = item.Spec.VolumeName
+				pvcInfo.CasType = casType
+				pvcInfo.StorageClassName = *item.Spec.StorageClassName
+
+				// fetching the underlying CStorVolume for the PV, to get the phase and size and notify the user
+				// if the CStorVolume is not found.
+				cv, err := clientset.GetcStorVolume(item.Spec.VolumeName)
+				if err != nil {
+					fmt.Println("Underlying CstorVolume is not found for: ", item.Name)
+				} else {
+					pvcInfo.Size = cv.Spec.Capacity.String()
+					pvcInfo.PVStatus = cv.Status.Phase
+				}
+
+				// fetching the underlying CStorVolumeConfig for the PV, to get the cvc info and Pool Name and notify the user
+				// if the CStorVolumeConfig is not found.
+				cvc, err := clientset.GetCVC(item.Spec.VolumeName)
+				if err != nil {
+					fmt.Println("Underlying CstorVolumeConfig is not found for: ", item.Name)
+				} else {
+					pvcInfo.Pool = cvc.Labels[cstortypes.CStorPoolClusterLabelKey]
+					cvcInfo.Name = cvc.Name
+					cvcInfo.ReplicaCount = len(cvc.Status.PoolInfo)
+					cvcInfo.PoolInfo = cvc.Status.PoolInfo
+					cvcInfo.Version = cvc.VersionDetails.Status.Current
+					cvcInfo.Upgrading = !(cvc.VersionDetails.Status.Current == cvc.VersionDetails.Desired)
+				}
+
+				// fetching the underlying CStorVolumeAttachment for the PV, to get the attached to node and notify the user
+				// if the CStorVolumeAttachment is not found.
+				cva, err := clientset.GetCVA(item.Spec.VolumeName)
+				if err != nil {
+					fmt.Println("Underlying CstorVolumeAttachment is not found for: ", item.Name)
+				} else {
+					pvcInfo.AttachedToNode = cva.Spec.Volume.OwnerNodeID
+				}
+
+				// fetching the underlying CStorVolumeReplicas for the PV, to list their details and notify the user
+				// none of the replicas are running if the CStorVolumeReplicas are not found.
+				cvrs, err := clientset.GetCVR(item.Spec.VolumeName)
+				if err == nil && len(cvrs.Items) > 0 {
+					pvcInfo.Used = client.GetUsedCapacityFromCVR(cvrs)
+				}
+
+				// Printing the Filled Details of the Cstor PVC
+				err = util.PrintByTemplate("pvc", cstorPvcInfoTemplate, pvcInfo)
+				if err != nil {
+					return err
+				}
+
+				// fetching the underlying TargetPod for the PV, to display its relevant details and notify the user
+				// if the TargetPod is not found.
+				targetPod, err := clientset.GetCstorVolumeTargetPod(item.Spec.VolumeName)
+				if err == nil {
+					targetPodOutput := make([]string, 2)
+					fmt.Printf("Target Details :\n")
+					targetPodOutput[0] = "Namespace|Name|Ready|Status|Age|IP|Node"
```

Comment:
Should dropping the namespace from the output be a good idea, considering that the namespace is already provided in the flag or from the kube-context(it gets a bit tricky here)?

Reply:
- The namespace here refers to the namespace where the `openebs` is installed. Since we are not showing anything about `openebs` ns in the pvc describe, this has been kept to infer to the user that the targetpod is a part of `openebs` ns. If this seems to be redundant, we can remove it.

Reply:
Ah, ignore my suggestion. That'd be helpful to have.